### PR TITLE
Merged Users and Manager services

### PR DESCRIPTION
**Pull request by mchf**
## Problem

Purpose is to save some resources. As there is low benefit from having users in separate process, here it patched to run in same process as the manager service.

- https://trello.com/c/jtFrKeLQ/3348-3-agama-move-users-configuration-to-the-main-service

## Solution

*Short description of the fix.*

## Testing

- *Added a new unit test*
- *Tested manually*

## Screenshots

*If the fix affects the UI attach some screenshots here.*

### CODE REVIEW

**Comment by mchf** [comment]
I've tried to register different objects per Users1 and Agama1. However d-feet shows the same object for both of them

dbus_users

File: service/lib/agama/dbus/manager_service.rb [hunk 1]
```diff
@@ -70,7 +72,12 @@ def start
 
       # Exports the installer object through the D-Bus service
       def export
+        # users service initialization
+        users_service = bus.request_service("org.opensuse.Agama.Users1")
+
+        # manager service initialization
         dbus_objects.each { |o| service.export(o) }
+        users_service.export(Agama::DBus::Users.new(Agama::Users.new(logger), logger))
```
Comment (mvidner) [inline service/lib/agama/dbus/manager_service.rb:80]:
```suggestion
        service.export(Agama::DBus::Users.new(Agama::Users.new(logger), logger))
```

when I do this (which is an obscure workaround!)  it starts to work:

```console
ba3c5497c2c2:/ # abusctl tree org.opensuse.Agama.Users1
Only root object discovered.
```

Oops. Ruby-dbus has a race condition, exposing the service before you export any objects. Let's try again.

```console
ba3c5497c2c2:/ # abusctl tree org.opensuse.Agama.Users1
└─/org
  └─/org/opensuse
    ├─/org/opensuse/Agama
    │ └─/org/opensuse/Agama/Users1
    └─/org/opensuse/Agama1
      └─/org/opensuse/Agama1/Manager
```

File: service/lib/agama/dbus/manager_service.rb [hunk 2]
```diff
@@ -33,27 +35,38 @@ module DBus
     # It connects to the system D-Bus and answers requests on objects below
     # `/org/opensuse/Agama1`.
     class ManagerService
-      # Service name
+      # List of d-bus services exposed by this class
```
Comment (imobachgs) [inline service/lib/agama/dbus/manager_service.rb:38]: np: D-Bus

**Review by imobachgs — CHANGES_REQUESTED** [review]
It looks good. Please, add an entry to the `changes` file referencing this PR. Thanks!

**Comment by imobachgs** [comment]
Please, fix the conflicts too.

File: service/nohup.out [hunk 3]
```diff
@@ -0,0 +1,4 @@
+
```
Comment (imobachgs) [inline service/nohup.out:1]: It looks like a left-over 😅

File: service/package/rubygem-agama.changes [hunk 4]
```diff
@@ -1,3 +1,8 @@
+-------------------------------------------------------------------
+Wed Jun  7 08:54:58 UTC 2023 - Michal Filka <mfilka@suse.com>
+
+- Merged Users service into the main one to safe some resources
```
Comment (imobachgs) [inline service/package/rubygem-agama.changes:4]:
```suggestion
- Merge the users D-Bus service into the main one to save some resources
  (gh#openSUSE/agama#596).
```

**Review by imobachgs — COMMENTED** [review]
Almost LGTM :-) Just two small things.

File: service/nohup.out (hunk 3 above)
Comment (mchf) [reply service/nohup.out]: ups

**Review by imobachgs — APPROVED** [review]

**Pull request merged**